server/__test__/routes/auth.test.js

```diff
@@ -3,80 +3,136 @@ const app = require('../../app');
+      } = await request(app)
+        .post('/api/v1/signup')
+        .send({
+          ...userData,
+          activityId: '500',
+          email: 'notfoundemail@gmail.com',
+        });
+      expect(statusCode).toBe(404);
+      expect(message).toBe('activity id not found');
+    });
+
+    test('check successful sign-up, status should be 200 and (signed up successfully) message', async () => {
+      const {
+        body: { message },
+        statusCode,
+      } = await request(app)
+        .post('/api/v1/signup')
+        .set({
+          'Content-Type': 'application/json',
+        })
+        .send({
+          ...userData,
+          email: 'imans.ewaiti@gmail.com',
+        });
+      expect(statusCode).toBe(201);
+      expect(message).toBe('signed up successfully');
+    });
   });
 
-  test('check successful sign-up, status should be 200 and (signed up successfully) message', async () => {
-    const {
-      body: { message },
-      statusCode,
-    } = await request(app)
-      .post('/api/v1/signup')
-      .set({
-        'Content-Type': 'application/json',
-      })
-      .send({
-        ...userData,
-        email: 'imans.ewaiti@gmail.com',
-      });
-    expect(statusCode).toBe(201);
-    expect(message).toBe('signed up successfully');
+  describe('POST /api/v1/login', () => {
+    it('router returns 400 if there is no password', async () => {
+      const { statusCode } = await request(app)
+        .post('/api/v1/login')
+        .send({ email: 'user@gmail.com', password: '' });
+      expect(statusCode).toBe(400);
+    });
+
+    it('router returns 400 if there is no email', async () => {
+      const { statusCode } = await request(app)
+        .post('/api/v1/login')
+        .send({ email: '', password: 'hohoho123123' });
+      expect(statusCode).toBe(400);
+    });
+
+    it('router returns 200 if user logged in successfully', async () => {
+      const { statusCode } = await request(app)
+        .post('/api/v1/login')
+        .set({
+          'Content-Type': 'application/json',
+        })
+        .send(
+          JSON.stringify({
+            email: 'zein@gmail.com',
+            password: 'zein2002jendeya',
+          }),
+        );
+
+      expect(statusCode).toBe(200);
+    });
+    it('router returns 401 if password is incorrect', async () => {
+      const {
+        body: { message },
+        statusCode,
+      } = await request(app)
+        .post('/api/v1/login')
+        .set({
+          'Content-Type': 'application/json',
+        })
+        .send(
+          JSON.stringify({ email: 'zein@gmail.com', password: 'zein2002j' }),
+        );
+      expect(statusCode).toBe(401);
+      expect(message).toBe('invalid password');
+    });
+
+    test('router returns 401 if email is incorrect', async () => {
+      const { statusCode } = await request(app)
+        .post('/api/v1/login')
+        .send({ email: 'user@gmail.com', password: 'dddddd888' });
+      expect(statusCode).toBe(422);
```

Comment:
Hi! I think that for consistency, you could use the same error status code for incorrect password and incorrect email.  You could use 401 or 422 (or even 404, that email/password combination was not found), but I think it would be simpler if both types of incorrect data returned the same error status code.
It's not an urgent change that needs to be made, but something to keep in mind for next time.

Reply:
Really thanks for this suggestion
sure we will keep it in mind.

Reply:
afwan!